### PR TITLE
Remove Greaselion

### PR DESCRIPTION
**Pull request by atuchin-m**
Resolves https://github.com/brave/brave-browser/issues/40873

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: build/commands/lib/l10nUtil.js [hunk 1]
```diff
@@ -210,43 +210,6 @@ const l10nUtil = {
...
-
-    const greaselionRules = JSON.parse(jsonContent)
-    if (!greaselionRules) {
-      console.error('Malformed Greaselion.json')
-      return []
-    }
-
-    let paths = []
-    greaselionRules.forEach((rule) => {
-      if (rule.messages) {
```
Comment (atuchin-m) [inline build/commands/lib/l10nUtil.js:242]: @mkarolin @zenparsing FIY
I can't see any `messages` here: https://github.com/brave/brave-site-specific-scripts/blob/master/Greaselion.json

**Review by zenparsing — APPROVED** [review]

File: build/commands/lib/l10nUtil.js (hunk 1 above)
Comment (emerick) [reply build/commands/lib/l10nUtil.js:242]: We removed the scripts that used that feature, from what I recall. We can probably remove the Greaselion section from Transifex if it's still there @mkarolin.

File: build/commands/lib/pushL10n.js [hunk 2]
```diff
@@ -11,14 +16,7 @@ const pushL10n = (options) => {
...
   if (options.extension) {
-    const extensionPath = options.extension_path
-    if (options.extension === 'greaselion') {
-      l10nUtil.getGreaselionScriptPaths(extensionPath).forEach((sourceStringPath) => {
-        util.run('python', ['script/push-l10n.py', '--source_string_path', sourceStringPath], cmdOptions)
-      })
-      return
-    }
-    console.error('Unknown extension: ', options.extension)
+    console.error('No extension is supported now')
```
Comment (emerick) [inline build/commands/lib/pushL10n.js:19]: Is there any reason not to just drop this option entirely?

**Review by emerick — APPROVED** [review]

**Review by antonok-edm — APPROVED** [review]

**Review by iefremov — APPROVED** [review]

**Review by bridiver — APPROVED** [review]

File: build/commands/lib/pullL10n.js [hunk 3]
```diff
@@ -7,14 +12,7 @@ const pullL10n = (options) => {
...
-    const extensionPath = options.extension_path
-    if (options.extension === 'greaselion') {
-      l10nUtil.getGreaselionScriptPaths(extensionPath).forEach((sourceStringPath) => {
-        util.run('python', ['script/pull-l10n.py', '--source_string_path', sourceStringPath], cmdOptions)
-      })
-      return
-    }
-    console.error('Unknown extension: ', options.extension)
+    console.error('No extension is supported now')
     process.exit(1)
```
Comment (mkarolin) [inline build/commands/lib/pullL10n.js:16]: Can just delete all of this and the extension option.

File: build/commands/lib/pushL10n.js (hunk 2 above)
Comment (mkarolin) [reply build/commands/lib/pushL10n.js:19]: Agree, let's just drop the whole thing.

File: build/commands/lib/pullL10n.js [hunk 4]
```diff
@@ -1,4 +1,9 @@
-const path = require('path')
+/* Copyright (c) 2017 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+ const path = require('path')
```
Comment (mkarolin) [inline build/commands/lib/pullL10n.js:6]: Extra leading space.

**Review by mkarolin — APPROVED** [review]
l10n scripts changes LGTM

**Pull request merged**